File: PgAce.App/MainForm.cs
using System;
using System.Data;
using System.Linq;
using System.Windows.Forms;
using WeifenLuo.WinFormsUI.Docking;
using PgAce.App.Views;

namespace PgAce.App
{
    public class MainForm : Form
    {
        private readonly DockPanel _dockPanel;
        private readonly ResultGridView _resultGrid;

        public MainForm()
        {
            Text = "PgAce";
            Width = 800;
            Height = 600;

            _dockPanel = new DockPanel();
            _dockPanel.Dock = DockStyle.Fill;
            Controls.Add(_dockPanel);

            var dbExplorer = new DbExplorerView();
            dbExplorer.Show(_dockPanel, DockState.DockLeft);

            var sqlEditor = new SqlEditorView();
            sqlEditor.Show(_dockPanel, DockState.Document);

            var sqlExplorer = new SqlExplorerView();
            sqlExplorer.Show(_dockPanel, DockState.DockRight);

            _resultGrid = new ResultGridView();
            _resultGrid.Show(_dockPanel, DockState.DockBottom);

            var properties = new PropertiesView();
            properties.Show(_resultGrid.Pane, DockAlignment.Right, 0.5);

            var searchView = new SearchView();
            searchView.SearchRequested += OnSearchRequested;
            searchView.Show(_dockPanel, DockState.DockTop);
        }

        private void OnSearchRequested(string query)
        {
            var items = new[]
            {
                new { Id = 1, Name = "Apple" },
                new { Id = 2, Name = "Banana" },
                new { Id = 3, Name = "Cherry" },
                new { Id = 4, Name = "Date" }
            };

            var filtered = items.Where(i => i.Name.Contains(query, StringComparison.OrdinalIgnoreCase));

            var table = new DataTable();
            table.Columns.Add("Id", typeof(int));
            table.Columns.Add("Name", typeof(string));

            foreach (var item in filtered)
            {
                table.Rows.Add(item.Id, item.Name);
            }

            _resultGrid.SetResults(table);
        }
    }
}

File: PgAce.App/MainForm.vb
Imports System
Imports System.IO
Imports System.Windows.Forms
Imports WeifenLuo.WinFormsUI.Docking
Imports PgAce.App.Forms

Namespace PgAce.App
    Public Class MainForm
        Inherits Form

        Private ReadOnly _dockPanel As New DockPanel()
        Private ReadOnly _menuStrip As New MenuStrip()
        Private ReadOnly _statusStrip As New StatusStrip()
        Private Const LayoutFile As String = "layout.xml"

        Public Sub New()
            Text = "PgAce"
            Width = 1024
            Height = 768

            _dockPanel.Dock = DockStyle.Fill
            Controls.Add(_dockPanel)

            ' menu
            Dim fileMenu = New ToolStripMenuItem("&File")
            Dim exitItem = New ToolStripMenuItem("E&xit", Nothing, Sub(s, e) Close())
            fileMenu.DropDownItems.Add(exitItem)
            _menuStrip.Items.Add(fileMenu)
            MainMenuStrip = _menuStrip
            Controls.Add(_menuStrip)

            Controls.Add(_statusStrip)

            AddHandler Load, AddressOf MainForm_Load
            AddHandler FormClosing, AddressOf MainForm_FormClosing
        End Sub

        Private Sub MainForm_Load(sender As Object, e As EventArgs)
            If File.Exists(LayoutFile) Then
                Try
                    _dockPanel.LoadFromXml(LayoutFile, AddressOf DeserializeDockContent)
                Catch
                    ShowDefaultLayout()
                End Try
            Else
                ShowDefaultLayout()
            End If
        End Sub

        Private Sub MainForm_FormClosing(sender As Object, e As FormClosingEventArgs)
            _dockPanel.SaveAsXml(LayoutFile)
        End Sub

        Private Function DeserializeDockContent(persistString As String) As IDockContent
            Select Case persistString
                Case NameOf(DbExplorerPane), "PgAce.App.Forms.DbExplorerPane"
                    Return New DbExplorerPane()
                Case NameOf(SqlEditorPane), "PgAce.App.Forms.SqlEditorPane"
                    Return New SqlEditorPane()
                Case NameOf(ResultPane), "PgAce.App.Forms.ResultPane"
                    Return New ResultPane()
                Case NameOf(PropertiesPane), "PgAce.App.Forms.PropertiesPane"
                    Return New PropertiesPane()
                Case NameOf(SqlExplorerPane), "PgAce.App.Forms.SqlExplorerPane"
                    Return New SqlExplorerPane()
            End Select
            Return Nothing
        End Function

        Private Sub ShowDefaultLayout()
            Dim dbExplorer = New DbExplorerPane()
            dbExplorer.Show(_dockPanel, DockState.DockLeft)

            Dim sqlExplorer = New SqlExplorerPane()
            sqlExplorer.Show(_dockPanel, DockState.DockLeft)

            Dim sqlEditor = New SqlEditorPane()
            sqlEditor.Show(_dockPanel, DockState.Document)

            Dim result = New ResultPane()
            result.Show(_dockPanel, DockState.DockBottom)

            Dim propertiesPane = New PropertiesPane()
            propertiesPane.Show(_dockPanel, DockState.DockRight)
        End Sub
    End Class
End Namespace
